packages/preferences-api/src/common/preference-proxy.ts

```diff
@@ -5,27 +5,33 @@
  * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
  */
 
-import { Disposable, DisposableCollection, Event, Emitter, Deferred } from '@theia/core/lib/common';
+import { Disposable, DisposableCollection, Event, Emitter } from '@theia/core/lib/common';
 import { PreferenceService, PreferenceChangedEvent } from "./preference-service";
 
 export type Configuration = {
     [preferenceName: string]: any
 }
 export type PreferenceEventEmitter<T> = {
     readonly onPreferenceChanged: Event<{
-        readonly preferenceName: keyof T
-        readonly newValue?: T[keyof T]
-        readonly oldValue?: T[keyof T]
-    }>
+        changes: [{
```

Comment:
@akosyakov if you have time could you check if the latest patch is what you meant with the rollback? Thanks